libraries/AP_BattMonitor/AP_BattMonitor_Backend.cpp

```diff
@@ -30,15 +30,16 @@ AP_BattMonitor_Backend::AP_BattMonitor_Backend(AP_BattMonitor &mon, AP_BattMonit
 {
 }
 
-/// capacity_remaining_pct - returns the % battery capacity remaining (0 ~ 100)
-uint8_t AP_BattMonitor_Backend::capacity_remaining_pct() const
+/// capacity_remaining_pct - returns true if the battery % is available and writes to the percentage argument
+bool AP_BattMonitor_Backend::capacity_remaining_pct(int8_t &percentage) const
 {
-    float mah_remaining = _params._pack_capacity - _state.consumed_mah;
-    if ( _params._pack_capacity > 10 ) { // a very very small battery
-        return MIN(MAX((100 * (mah_remaining) / _params._pack_capacity), 0), UINT8_MAX);
-    } else {
-        return 0;
+    // anything under 10 mAh is an invalid capacity and so will be our measurement of remaining capacity
+    if ( _params._pack_capacity <= 10) {
+        return false;
     }
+    float mah_remaining = _params._pack_capacity - _state.consumed_mah;
+    percentage = MIN(MAX((100 * (mah_remaining) / _params._pack_capacity), 0), UINT8_MAX);
```

Comment:
```suggestion
    percentage = MIN(MAX((100 * mah_remaining / _params._pack_capacity), 0), INT8_MAX);
```

I'd be a bit concerned about returning anything outside the 0-100 range here, actually, and would prefer 
```suggestion
    percentage = constrain_float(100 * mah_remaining / _params._pack_capacity, 0, 100);
```

Callers might not expect anything out of that range and might be trying to shove it into a datatype that won't take the larger range.  This appears to be a pre-existing problem.

Reply:
thanks. constrain_float() is a lot more readable, too.

But could limiting it to 100% affect vehicles that have battery charging built-in? solar panels or other kind of generators?
Seeing a percentage above 100% would be pretty obvious to me that the battery is being over-charged, while capping it at 100% could mask the issue.